core-client/src/main/java/org/glassfish/jersey/client/authentication/HttpAuthenticationFilter.java

```diff
@@ -247,7 +246,22 @@ public void filter(ClientRequestContext request, ClientResponseContext response)
     }
 
     private String getCacheKey(ClientRequestContext request) {
-        return request.getUri().toString() + ":" + request.getMethod();
+        URI requestUri = request.getUri();
```

Comment:
This method duplicates code from DigestAuthenticator#getCacheKey. Does it make sense to put the code to a new static method to a new utility AuthenticationUtil class in the package?

Reply:
Yes, asbolutely. Done.